keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt

```diff
@@ -158,6 +159,36 @@ class MemoryCloudDriverCache(
         ?: notFound("Subnet with purpose \"$purpose\" not found in $account:$region")
     }
 
+  private val certificatesByName: AsyncLoadingCache<String, Certificate> =
+    cacheFactory
+      .asyncLoadingCache("certificatesByName") { name ->
+        runCatching {
+          cloudDriver
+            .getCertificates()
+            .find { it.serverCertificateName == name }
+            ?.also {
+              certificatesByArn.put(it.arn, completedFuture(it))
+            }
+        }
+          .handleNotFound()
+          ?: notFound("Certificate with name $name not found")
+      }
```

Comment:
New caches for certs. This pattern is a bit screwy for endpoints like this that return all the data in a single payload. We're doing the same thing for accounts, subnets, AZs, etc. I'm going to try to pull together another PR with a better approach.